axelor-sale/src/main/java/com/axelor/apps/sale/service/cart/CartProductServiceImpl.java

```diff
@@ -57,23 +58,24 @@ public void addToCart(Product product) throws AxelorException {
     }
     CartLine cartLine = cartLineRetrievalService.getCartLine(cart, product);
     if (cartLine == null) {
-      cartLineCreateService.createCartLine(cart, product);
+      cartLineCreateService.createCartLine(cart, product, null);
     } else {
-      cartLineUpdateService.updateCartLine(cartLine);
+      cartLineUpdateService.updateCartLine(cartLine, null);
```

Comment:
Instead of putting null as a parameter here, you can create another new method createCartLine/updateCartline without the qty which will call the base method with a null qty. This should make the code more readable.